hack/lib/protoc.sh

```diff
@@ -27,7 +27,8 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 # $1: Full path to the directory where the api.proto file is
 function kube::protoc::generate_proto() {
   kube::golang::setup_env
-  GO111MODULE=on GOPROXY=off go install k8s.io/code-generator/cmd/go-to-protobuf/protoc-gen-gogo
+  KUBE_VERBOSE=-1 hack/make-rules/build.sh \
+      k8s.io/code-generator/cmd/go-to-protobuf/protoc-gen-gogo
```

Comment:
huh... is changing from `go install` to `hack/make-rules/build.sh` for our tools that don't need any CGO/platform/static-compilation logic an improvement? the former is much easier to understand, imo

the GO111MODULE/GOPROXY vars are ugly, but also make it super clear we're building in module mode and not fetching any content from the network...

Reply:
hack/make-rules/build.sh sets all sorts of things that may or not matter for these "build tools".  I can go wither way.  As part of go workspaces, I was assuming build.sh was a better destination, since it ensures that the build output is local (_output/bin, local cache, etc.).  If any tool-script forgets to call `golang::setup_env` it will splat things into the users own GOPATH, which is a) obnoxious and b) hard to debug when there are `client-gen` (etc) binaries in your own GOPATH and the local _output.

With go workspaces, vendoring is not supported, so we need to figure out how we want to do that - one option is to commit a full module cache, which requires that we set GOMODCACHE.

I agree that `go install` is simpler and more obvious.  `GO111MODULE=on` will not be needed on workspaces lands.  `GOPROXY=off` is not compatible with workspaces.  So we will end up with "naked" `go install` which is nice - except for the MODCACHE stuff, which maybe we can fix in different ways?

I can go either way - drop this PR or keep?

Reply:
if there's env stuff we need to set up (like GOBIN/GOPATH/GOMODCACHE), that could make sense in `kube::golang::setup_env` and still let the install use the standard command.

I'd love to move in the direction of standard go commands (go test, go install, etc) being sufficient for most of what we need to do, so my inclination would be to try to use those if we can. That said, I don't feel super strongly either way.

Reply:
I tagged https://github.com/kubernetes/kubernetes/pull/115222 with a hold, if you're ok going that direction